src/decode.rs

```diff
@@ -133,7 +133,13 @@ fn decode_load(i: u32) -> DecodingResult {
 fn decode_op_imm(i: u32) -> DecodingResult {
     match (i >> 12) & 0b111 {
         0b000 => Ok(Instruction::Addi(IType(i))),
+        0b010 => Ok(Instruction::Slti(IType(i))),
+        0b011 => Ok(Instruction::Sltiu(IType(i))),
+        0b100 => Ok(Instruction::Xori(IType(i))),
+        0b110 => Ok(Instruction::Ori(IType(i))),
         0b111 => Ok(Instruction::Andi(IType(i))),
+        0b001 => Ok(Instruction::Slli(IType(i))),
```

Comment:
I think we should ensure the bits [31:26] for `slli`, `srli`, and `srai` are all set correctly.